client/src/components/UserInput/UserInput.jsx

```diff
@@ -0,0 +1,51 @@
+import React from "react";
+import PropTypes from "prop-types";
+import "./userInput.css";
+
+import { TextField } from "monday-ui-react-core";
+import { Add } from "monday-ui-react-core/dist/allIcons";
+import "monday-ui-react-core/dist/main.css";
+
+export const UserInput = ({
+    loading,
+    onIconClick,
+    value,
+    onChange,
+    updateApp,
+}) => {
+    const [task, setTask] = React.useState(value);
+    return (
+        <>
+            <TextField
+                placeholder={"Add a new task"}
+                value={task}
+                iconName={Add}
+                loading={loading}
+                onChange={(value) => {
+                    setTask(value);
+                    onChange(value);
+                }}
+                onIconClick={async () => {
+                    await onIconClick(task);
```

Comment:
it's better to extract this logic to a method and call it here

Reply:
I'm injecting from taskContainerApp to this comp, I practice on building a stateless components